docs/docsite/rst/desired_state_schema.rst

```diff
@@ -0,0 +1,61 @@
+.. _ansible_collections.cloud.pravic.docsite.desired_state:
+
+YAML specification for defining desired resource state
+======================================================
+
+The users who intend to use the `cloud.pravic Collection <https://github.com/ansible-collections/pravic>`_ to provision/deprovision Cloud resources will need to use this YAML specification for defining desired resource state:
+
+```yaml
+  ResourceGroups:
+  description: Array of resource groups.
+  type: array
+  items:
+    type: object
+    properties:
+      ResourceGroupName:
+        description: The unique identifier for a group of resources. ResourceGroupName should not include any spaces.
+        type: string
+      Resources:
+        description: Dictionary of resources belonging to ResourceGroupName.
+        type: object
+        properties:
+          name:
```

Comment:
I'm assuming this is JSON Schema, in which case, isn't this saying that a resource needs a property literally called `name`? I think we would want to use `patternProperties` here, wouldn't we?

Reply:
It should be the YAML definition schema.

Reply:
The only YAML schema I'm aware of is based on JSON schema. Is there something else?

Reply:
I took it from here https://asdf-standard.readthedocs.io/en/1.5.0/schemas/yaml_schema.html

Reply:
What I meant is something like this:
```
- ResourceGroupName: cloud_1
    Resources:
    bucket_02:
      Type: AWS::S3::Bucket
```
where that name is literally bucket_02.